octoprint_mrbeam/static/js/app/view-models/modal/laserhead-changed.js

```diff
@@ -59,19 +59,31 @@ $(function () {
         };
 
         self.onUserLoggedIn = function () {
-            if (self.loginState.currentUser?.()?.active) {
-                if (self.settings.settings.plugins.mrbeam.laserheadChanged()) {
-                    self.laserheadModelId(
-                        self.settings.settings.plugins.mrbeam.laserhead.model_id()
-                    );
-                    self.laserheadModelSupported(
-                        self.settings.settings.plugins.mrbeam.laserhead.model_supported()
-                    );
-                    $("#laserhead_changed").modal("show");
-                }
+            if (self.isLaserheadChanged()) {
+                self.laserheadModelId(
+                    self.settings.settings.plugins.mrbeam.laserhead.model_id()
+                );
+                self.laserheadModelSupported(
+                    self.settings.settings.plugins.mrbeam.laserhead.model_supported()
+                );
+                $("#laserhead_changed").modal("show");
             }
         };
 
+        self.isLaserheadChanged = function () {
+            return (
+                self.loginState.currentUser?.()?.active &&
+                self.settings.settings.plugins.mrbeam.laserheadChanged()
+            );
+        };
+
+        self.laserheadXDetectedForFirstTime = function () {
```

Comment:
This is only checking if the laserhead was changed to a laserhead x right? It doesn't contain information about the first time this happened or am I wrong? I would then propose to rename the function. I would vote to not overcomplicate it and leave it with this logic without taking into account if it's the first time it was changed or not but just if it was changed to laserhead x from any different laserhead

Reply:
I will change to this:
```suggestion
        self.laserheadXDetected = function () {
```

Reply:
Done